### ISSUE TITLE
the bp of binary_conv layer

### ISSUE THREAD

**Issue by NHZlX** [HongxingGao/XNOR-Net#2]
I take a cursory look at the implementation of the binary_conv_layer , and i find that the bp process of the weight of the layer  is inconsistent with the paper. This is for simplicity or mistake, i will try the effect.

**Comment by HongxingGao** [comment]
Yes,  I have noticed that. I have implemented the gradient calculation as specified in the paper.  But the final result did not show any improvement on ImageNet.  Hence,  i didn't  update that code.

**Comment by NHZlX** [comment]
@HongxingGao  I am trying to modify it, is it convenient for you to give me your revised code?
zlx_hg@163.com

**Comment by caseyanya** [comment]
I'm also suffering from accuracy on ImageNet classification (darknet19 in binary weight from scratch). 
What is your Top-1 accuracy?
I cannot even pass 20% Top-1 accuracy using binary weight only.
